hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/RegionHDFSBlockLocationFinder.java

```diff
@@ -60,6 +63,9 @@
   private static final Logger LOG = LoggerFactory.getLogger(RegionHDFSBlockLocationFinder.class);
   private static final long CACHE_TIME = 240 * 60 * 1000;
   private static final float EPSILON = 0.0001f;
+  // The max size of the cache,
+  private static final String BALANCER_REGIONBLOCKLOCATION_CACHE_MAX_SIZE =
+    "hbase.master.balancer.regionBlockLocation.cache.max.size";
```

Comment:
The purpose to limit this cache size is to optimize the GC time?

Reply:
Just give the cache an upper limit, you can see details at https://issues.apache.org/jira/browse/HBASE-25624